src/main/scala/ExportConsumerMetricsToRegistryActor.scala

```diff
@@ -0,0 +1,79 @@
+import KafkaClientActor.{DescribeKafkaClusterConsumer, Command, ListConsumers}
+import akka.actor._
+import akka.pattern.ask
+import akka.stream.ActorMaterializer
+import akka.util.Timeout
+import com.codahale.metrics.Gauge
+import nl.grons.metrics.scala.{ActorInstrumentedLifeCycle, ReceiveCounterActor, ReceiveExceptionMeterActor, ReceiveTimerActor}
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+
+object ExportConsumerMetricsToRegistryActor {
+  def props(kafkaClientActorRef: ActorRef)(implicit actorSystem: ActorSystem, materializer: ActorMaterializer) =
+    Props(classOf[ExportConsumerMetricsToRegistryActor], kafkaClientActorRef, actorSystem, materializer)
+}
+
+class BaseExportConsumerMetricsToRegistryActor(kafkaClientActorRef: ActorRef)
+                                              (implicit actorSystem: ActorSystem, materializer: ActorMaterializer)
+  extends Actor
+    with ActorLogging
+    with nl.grons.metrics.scala.DefaultInstrumented
+    with ActorInstrumentedLifeCycle {
+
+  implicit val timeout = Timeout(60 seconds)
+  implicit val apiExecutionContext = actorSystem.dispatchers.lookup("exporter-dispatcher")
+
+  private def askFor[RES](command: Command)(implicit tag: ClassTag[RES]) =
+    (kafkaClientActorRef ? command).mapTo[RES]
+
+  def receive = {
+    case _ =>
+      log.info("Exporting lag info to metrics registry!")
+      val consumerList = askFor[List[String]](ListConsumers)
+      consumerList.map(_.foreach(consumerGroup => {
+        val groupInfo = askFor[GroupInfo](DescribeKafkaClusterConsumer(consumerGroup))
+        groupInfo.map { gi =>
+          gi.partitionAssignmentStates.map(pa => {
+            pa.map { p =>
```

Comment:
Does it make sense to send all the per partition information?
Utilizing this for say autoscaling it might make more sense to have a single metric - max_lag or average_lag.

Reply:
You could always use the graphs to aggregate the data together if they don't need such fine detail.

Since this is coming from Buffalo, can you ask a member to review? That way we can tailer it to their needs